### ISSUE TITLE
[Question] Render custom hotspots into canvas

### ISSUE THREAD

**Issue by ondrej-vipo** [google/marzipano#280]
Hello,
Is there any way to render custom "hotspots" into canvas itself via webgl?

**Comment by tjgq** [comment]
There isn't, sorry. We decided to implement hotspots using DOM elements because they're more versatile (you can render any DOM content, not just images).

Can you explain a little bit more about your use case, and why DOM-based hotspots aren't suitable?

**Comment by ondrej-vipo** [comment]
@tjgq Hello, I would like to display panoramas with hotspots in VR and my understanding of WebVR is that only stuff from canvas is beeing displayed on VR screen.

**Comment by tjgq** [comment]
Yes, that's a WebVR limitation; all it does is display a canvas, there is no DOM.

The point of hotspots is interactivity; if you're on WebVR all you can display is static content, is it not? At that point, why not pre-render the hotspots on top of the panorama?

**Comment by jordyno** [comment]
@tjgq what are the benefits of using WebVR as opposed to side-by-side demo? The side-by-side version seems to be supporting all of the features. Is there a major performance boost with WebVR? Or compatibility?

**Comment by ondrej-vipo** [comment]
@tjgq  Well thats the thing: I wanted to set/unset the hotspots based on server response that I can get via ajax call.

**Comment by ondrej-vipo** [comment]
@jordyno I think the only way is to use WebVR for devices like HTC Vive

**Comment by romangrinev** [comment]
> Can you explain a little bit more about your use case, and why DOM-based hotspots aren't suitable?

Another reason to render hotspots into canvas is to may polygonal hotspots. I was wondering can create a canvas layer and draw polygonal hotspots there using canvas functions.
